File: sdks/typescript/src/transport.ts
/**
 * Synap TypeScript SDK - Transport facade (backward-compatible re-exports)
 *
 * This file is the original transport entry point. All implementations have
 * been moved to `./transports/` for better modularity. This module re-exports
 * everything so existing imports (`from './transport'`) continue to work.
 */

export type { TransportMode } from './types';

export {
  SynapRpcTransport,
  toWireValue,
  fromWireValue,
} from './transports/synap-rpc';

export type { WireValue } from './transports/synap-rpc';

export { Resp3Transport } from './transports/resp3';

export { mapCommand, mapResponse } from './transports/command-map';
export type { MappedCommand } from './transports/command-map';

File: sdks/python/synap_sdk/transport.py
"""Binary TCP transports for Synap SDK.

This module re-exports the public API from the focused sub-modules:

- :mod:`synap_sdk.transport_rpc`   — SynapRPC (MessagePack-framed binary)
- :mod:`synap_sdk.transport_resp3` — RESP3 (Redis-compatible text protocol)
- :mod:`synap_sdk.command_map`     — command/response mapper

Wire encoding mirrors Rust's ``rmp_serde`` externally-tagged enum format:

=============  ==========================
Python value   Wire envelope
=============  ==========================
``None``       bare msgpack string "Null"
``str``        ``{"Str": "…"}``
``int``        ``{"Int": n}``
``float``      ``{"Float": f}``
``bool``       ``{"Bool": b}``
``bytes``      ``{"Bytes": b}``
``list``       ``{"Array": […]}``
=============  ==========================

Structs (Request, Response) are encoded as msgpack arrays.
"""

from __future__ import annotations

from typing import Literal

# Re-export transport implementations
from synap_sdk.transport_rpc import SynapRpcTransport, _from_wire, _to_wire
from synap_sdk.transport_resp3 import Resp3Transport

# Re-export command / response mapper
from synap_sdk.command_map import map_command as _map_command_public
from synap_sdk.command_map import map_command_optional as _map_command
from synap_sdk.command_map import map_response as _map_response

# ── Transport mode ─────────────────────────────────────────────────────────────

TransportMode = Literal["synaprpc", "resp3", "http"]

# Public aliases (used by client.py and external code)
map_command = _map_command
map_response = _map_response

__all__ = [
    "TransportMode",
    "SynapRpcTransport",
    "Resp3Transport",
    # private helpers re-exported for tests and introspection
    "_to_wire",
    "_from_wire",
    "_map_command",
    "_map_response",
    # public aliases
    "map_command",
    "map_response",
]
